src/controllers/auth.ts

```diff
@@ -0,0 +1,235 @@
+import { NextFunction, Request, Response } from "express";
+import userModel, { IUser } from "../models/user";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import { Document } from "mongoose";
+import { refreshTokenExpires, tokenExpires, tokenSecret } from "../config";
+
+// eslint-disable-next-line max-statements
+const register = async (req: Request, res: Response) => {
+    try {
+        const existingUserByEmail = await userModel.findOne({ email: req.body.email });
+        if (existingUserByEmail) {
+            const CONFLICT_STATUS = 409;
+            res.status(CONFLICT_STATUS).send("Email already in use");
+            return;
+        }
+
+        const existingUserByUsername = await userModel.findOne({ username: req.body.username });
+        if (existingUserByUsername) {
+            const CONFLICT_STATUS = 409;
+            res.status(CONFLICT_STATUS).send("Username already in use");
+            return;
+        }
+
+        const { password } = req.body;
+        const SALT_ROUNDS = 10;
+        const salt = await bcrypt.genSalt(SALT_ROUNDS);
+
+        const hashedPassword = await bcrypt.hash(password, salt);
+        const user = await userModel.create({
+            email: req.body.email,
+            password: hashedPassword,
+            username: req.body.username,
+        });
+        res.status(200).send({
+            _id: user._id,
+            email: user.email,
+            username: user.username,
+        });
+    } catch (err: any) {
+        const INTERNAL_SERVER_ERROR = 500;
+        res.status(INTERNAL_SERVER_ERROR).send("Server Error");
+    }
+};
+
+interface tTokens {
+    accessToken: string
+    refreshToken: string
+}
+
+const generateToken = (userId: string): tTokens | null => {
+    if (!tokenSecret) {
```

Comment:
This check is unnecessary and should be here.
If there is no tokenSecret the app should not start at all.
Therefore, this function should always return tokens.